internal/services/dataprotection/data_protection_backup_policy_mysql_flexible_server_resource.go

```diff
@@ -0,0 +1,697 @@
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"absolute_criteria": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForAbsoluteMarker(), false),
+								},
+
+								"days_of_week": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForDayOfWeek(), false),
+									},
+								},
+
+								"months_of_year": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForMonth(), false),
+									},
+								},
+
+								"scheduled_backup_times": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.IsRFC3339Time,
+									},
+								},
+
+								"weeks_of_month": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringInSlice(backuppolicies.PossibleValuesForWeekNumber(), false),
+									},
+								},
+							},
+						},
+					},
+
+					"life_cycle": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"data_store_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										// Confirmed with the service team that currently only `VaultStore` is supported.
+										// However, considering that `ArchiveStore` will be supported in the future, it would be exposed for user specification.
+										string(backuppolicies.DataStoreTypesVaultStore),
+									}, false),
+								},
+
+								"duration": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ForceNew:     true,
+									ValidateFunc: azValidate.ISO8601Duration,
+								},
+							},
+						},
+					},
+
+					"priority": {
+						Type:     pluginsdk.TypeInt,
+						Required: true,
+						ForceNew: true,
+					},
+				},
+			},
+		},
+
+		"time_zone": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
can we validate this?

Reply:
The validation is added per the existing [validation](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/maintenance/validate/maintenance.go).